### ISSUE TITLE
`bugprone-assignment-in-if-condition` false positive when lambda is passed to call in if-clause

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#56729]
```cpp
#include <algorithm>
#include <vector>

static void f()
{
    std::vector<const void*> members;
    if (std::any_of(members.begin(), members.end(), [&](const void* tok2) {
        {
            const void* dotTok;
            if (tok2)
                dotTok = nullptr; // warning
            else
                dotTok = nullptr; // warning
        }
        return true;
    })) 
    {
    }
}
```

```console
<source>:11:17: warning: an assignment within an 'if' condition is bug-prone [bugprone-assignment-in-if-condition]
                dotTok = nullptr; // warning
                ^
<source>:11:17: note: if it should be an assignment, move it out of the 'if' condition
<source>:11:17: note: if it is meant to be an equality check, change '=' to '=='
<source>:13:17: warning: an assignment within an 'if' condition is bug-prone [bugprone-assignment-in-if-condition]
                dotTok = nullptr; // warning
                ^
<source>:13:17: note: if it should be an assignment, move it out of the 'if' condition
<source>:13:17: note: if it is meant to be an equality check, change '=' to '=='
```
https://godbolt.org/z/xannPb43M

The warnings go away when you move the call out of the if-clause.
